compiler/src/dmd/mars.d

```diff
@@ -1937,10 +1948,41 @@ bool createModules(ref Strings files, ref Strings libmodules, ref Param params,
         modules.push(m);
         if (firstmodule)
         {
-            global.params.objfiles.push(m.objfile.toChars());
+            params.objfiles.push(m.objfile.toChars());
             firstmodule = false;
         }
     }
+
+    // Special module representing `stdin`
+    if (params.readStdin)
+    {
+        Module m;
+        if (createModule("__stdin.d", libmodules, params, target, eSink, m))
+            return true;
+        if (m is null)
+            return false;
+
+        modules.push(m);
+
+        // Set the source file contents of the module
+        OutBuffer buf;
+        buf.readFromStdin();
+        buf.write32(0);
+
+        const length = buf.length;
+        m.src = cast(ubyte[])(buf.extractSlice()[0 .. length - 4]);
```

Comment:
I think this adding 4 byte sentinel at the end is unneeded. There should be ample space at the end from readFromStdin -> small simplification win.